dimod/variables.py

```diff
@@ -70,27 +72,50 @@ def iter_deserialize_variables(variables):
     yield from map(deserialize_variable, variables)
 
 
-class Variables(cyVariables, abc.Set, abc.Sequence):
+class Variables(cyVariables, typing.AbstractSet[Variable], typing.Sequence[Variable]):
     """Set-like and list-like variables tracking.
 
     Args:
-        iterable (iterable):
+        iterable (iterable[:class:`~dimod.typing.Variable`], optional):
             An iterable of labels. Duplicate labels are ignored. All labels
             must be hashable.
 
+    Examples:
+
+        The variables object can be used to encode an ordered set of variables.
+
+        >>> variables = dimod.variables.Variables(['a', 'b', 0, 1])
+        >>> print(variables)
+        Variables(['a', 'b', 0, 1])
+
+        The variables object can be interacted with like a
```

Comment:
Perhaps something like, 
"The variables object supports a similar interface to the Python :class:`~collections.abc.Sequence` class."